### PR TITLE
Initial commit of dockerized KDK

### PR DESCRIPTION
**Pull request by rtluckie**
implements https://github.com/cisco-sso/k8s-devkit/issues/3

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        bc \
+        bind \
+        bind-utils \
+        colordiff \
+        ctags \
+        curl-devel \
+        device-mapper-persistent-data \
+        docker-ce \
+        dos2unix \
+        emacs \
```
Comment (dcwangmit01) [inline docker/Dockerfile:25]: I think we can avoid the X11 libs by: yum install emacs-nox

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        lua-devel \
+        luajit \
+        luajit-devel \
+        lvm2 \
+        most \
+        ncurses \
+        ncurses-devel \
+        nmap \
+        openssl-devel \
+        perl \
```
Comment (dcwangmit01) [inline docker/Dockerfile:47]: what requires Perl?

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        tcl-devel \
+        telnet \
+        tmux2u \
+        traceroute \
+        tree \
+        wget \
+        which \
+        xcb-util \
+        xdotool \
+        xorg-x11-font \
```
Comment (dcwangmit01) [inline docker/Dockerfile:76]: If we replace emacs with "emacs-nox" do we need these xorg libs?

Also, why are we explicitly specifying the xorg libs?  Is centos dep manager so broken?

File: docker/LICENSE [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (dcwangmit01) [inline docker/LICENSE:190]: We should set this line, in our other repos as well.

File: docker/files/provision-user [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+useradd ${KDK_USERNAME} -m -G wheel -s "${KDK_SHELL}"
+chown ${KDK_USERNAME} /home/${KDK_USERNAME}
+runuser -l ${KDK_USERNAME} -c "yadm clone --bootstrap ${KDK_DOTFILES_REPO}"
```
Comment (dcwangmit01) [inline docker/files/provision-user:6]: I don't see "yadm" installed anywhere on the docker image.

**Review by dcwangmit01 — APPROVED** [review]
Generally looks good.  I'd say merge it.  All of my comments are SHOULD have instead of MUST have.

File: docker/Dockerfile (hunk 3 above)
Comment (rtluckie) [reply docker/Dockerfile:76]: some of the xorg packages may not be required.  included while I was trying to get powerline fonts to render correctly.  should be cleaned up

File: docker/Dockerfile (hunk 2 above)
Comment (rtluckie) [reply docker/Dockerfile:47]: These packages are required when building vim from source with lua interpreter (which is required by a few common plugins).  We could probably remove them after vim is compiled.

File: docker/Dockerfile (hunk 1 above)
Comment (rtluckie) [reply docker/Dockerfile:25]: I'll test that out in followup PR.

File: docker/LICENSE (hunk 4 above)
Comment (rtluckie) [reply docker/LICENSE:190]: good call

File: docker/Dockerfile (hunk 1 above)
Comment (dcwangmit01) [reply docker/Dockerfile:25]: Followup PR is good.

File: docker/Dockerfile (hunk 2 above)
Comment (dcwangmit01) [reply docker/Dockerfile:47]: Nah, leave it.

File: docker/files/provision-user (hunk 5 above)
Comment (rtluckie) [reply docker/files/provision-user:6]: good catch.  typo in yum install.

**Pull request merged**